src/edu/csus/ecs/pc2/shadow/RemoteEventFeedMonitor.java

```diff
@@ -491,26 +491,42 @@ public void run() {
                                     }
                                     log.log(Level.INFO, "Found judgement event: " + event);
 
-                                    //process a judgement event
+                                    //process a judgment event
                                     try {
                                         //get a map of the data elements for the judgement
                                         Map<String, Object> judgementEventDataMap = (Map<String, Object>) eventMap.get("data");
-
-                                        // check if this is a "delete" event
                                         String operation = (String) eventMap.get("op");
-                                        if (operation != null && operation.equals("delete")) {
-                                            
-                                            //it is a delete; remove from the global map the judgement whose ID is specified
-                                            String idToDelete = (String) judgementEventDataMap.get("id");
-                                            synchronized (remoteJudgementsMapLock) {
-                                                getRemoteJudgementsMap().remove(idToDelete);
-                                            }
-
-                                            //TODO: how do we notify the local PC2 system that this judgement should be deleted??
-
+                                        String idToDelete = null;
+                                        boolean isDelete = false;
+                                        
+                                        // check if this is a "delete" event
+                                        // For the 2022-07 (and 2021-11) clics spec, "operation" will always be null since the "op" field was removed
+                                        // For the 2020-03 the "op" field will be non-null.  This is how we determine the feed type.                                        
```

Comment:
The code in question does not delete a run, rather, it deletes the judgement.  As such if we were to delete runs, we would have to process both delete "submissions" and delete "runs" events - we current do not do this.  Since we do not need this for WF, perhaps this should be rolled into a separate issue as described above for a deleted judgement.

I will, however, create a method to detect if an event is a delete event ("data":null) for new event feed, or op "delete" for the old.  I thought we said we are no longer supporting the old feed for the shadow, so perhaps remove the "compatibility" test for "op" propertery?  For now, it's harmless though.

Reply:
understood.